components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -5230,23 +5279,21 @@ def getFileInChunks(self, buf=2621440):
         :return:    Data from file in chunks
         :rtype:     Generator
         """
+        with self.asFileObj(buf) as f:
+            for chunk in f:
+                yield chunk
 
-        store = self._conn.createRawFileStore()
-        try:
-            store.setFileId(self._obj.id.val, self._conn.SERVICE_OPTS)
-            size = self._obj.size.val
-            if size <= buf:
-                yield store.read(0, long(size))
-            else:
-                for pos in range(0, long(size), buf):
-                    data = None
-                    if size-pos < buf:
-                        data = store.read(pos, size-pos)
-                    else:
-                        data = store.read(pos, buf)
-                    yield data
-        finally:
-            store.close()
+    def asFileObj(self, buf=2621440):
+        """
+        Return a read-only file-like object.
+        Caller must call close() on the file object after use.
+        This can be done automatically by using the object as a
+        ContextManager.
```

Comment:
Maybe to make this even clearer, provide a tiny example:
```
with  f.asFileObj() as fo:
    for chunk in fo:
        print chunk
```

Reply:
I've added an example